pkg/apis/stork/v1alpha1/volumeexport.go

```diff
@@ -0,0 +1,116 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// VolumeExportResourceName is name for the VolumeExport resource.
+	VolumeExportResourceName = "volumeexport"
+	// VolumeExportResourcePlural is the name for list of VolumeExport resources.
+	VolumeExportResourcePlural = "volumeexports"
+)
+
+// ExportType defines a method of achieving data transfer.
+type ExportType string
+
+const (
+	// ExportRsync means that data will be copied between two PVCs directly.
+	// Rsync supports both local and remote file copy.
+	ExportRsync ExportType = "rsync"
+	// ExportExtent means that some tool/service will handle data transfer.
+	ExportExtent ExportType = "extent"
+)
+
+// ExportStatus defines a status of VolumeExport..
+type ExportStatus string
+
+const (
+	// ExportStatusInitial is the status with with data transfer will start.
+	ExportStatusInitial ExportStatus = "Initial"
+	// ExportStatusPending when VolumeExport has not started yet.
+	ExportStatusPending ExportStatus = "Pending"
+	// ExportStatusInProgress when data is being transferred.
+	ExportStatusInProgress ExportStatus = "InProgress"
+	// ExportStatusFailed when data transfer is failed.
+	ExportStatusFailed ExportStatus = "Failed"
+	// ExportStatusSuccessful when data has been transferred.
+	ExportStatusSuccessful ExportStatus = "Successful"
+)
+
+// ExportStage defines different stages for VolumeExport when its Status changes
+// from Initial to Failed/Successful.
+type ExportStage string
+
+const (
+	// ExportStageInitial is the starting point for VolumeExport.
+	ExportStageInitial ExportStage = "Initial"
+	// ExportStageSourceSnapshot if the driver support this stage, it means a snapshot
+	// is being taken of the source PVC which will be used to transfer data with rsync.
+	ExportStageSourceSnapshot ExportStage = "SourceSnapshot"
```

Comment:
we can divide this into two stages and rename it to
- SnapshotScheduled
- SnapshotInProgress